### ISSUE TITLE
Provide a way to have a single reference which brings in both libraries and tools

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/cli#7670]
Right now, `PackageReference` items are generally for NuGet packages that have libraries, and `DotNetCliToolReference` items are used for CLI tools which can be invoked with `dotnet <toolname>`.

For some scenarios, there are both libraries to be referenced as well as tools that should be added to the project (for example, Entity Framework provides libraries, but could also provide tools for EF migrations or other scenarios).  In those cases, it would make sense to allow the tools and the libraries to be referenced with a single item.  Right now, you need a `PackageReference` as well as a `DotNetCliToolReference`, and while in an IDE like Visual Studio there's UI for adding a `PackageReference`, you have to edit the project file to manually add the `DotNetCliToolReference`.

**Comment by elsyms** [comment]
+1. 

Any chance we could also get a .NET CLI command to automatically add the Entity Framework tools? I was thinking something like

    dotnet add ef

It's a little annoying at the minute having to Google every time I need to add those command options (as I always forget!). It's such a common part of project initialization that I don't know why it isn't there. 

cc @livarcocc

**Comment by livarcocc** [comment]
I don't believe we want to specialize part of the the CLI like that.

@mlorbetske Is there a template that comes with EF on it? Should we consider adding one?

**Comment by RassulYunussov** [comment]
I agree, it is not very convenient to add a tool as a package and then rename it manually to DotNetCliToolReference
